c_glib/arrow-glib/reader.cpp

```diff
@@ -1497,6 +1488,387 @@ garrow_csv_reader_read(GArrowCSVReader *reader,
+#define GARROW_JSON_READ_OPTIONS_GET_PRIVATE(object) \
+  static_cast<GArrowJSONReadOptionsPrivate *>(       \
+    garrow_json_read_options_get_instance_private(   \
+      GARROW_JSON_READ_OPTIONS(object)))
+
+static void
+garrow_json_read_options_dispose(GObject *object)
+{
+  auto priv = GARROW_JSON_READ_OPTIONS_GET_PRIVATE(object);
+
+  if (priv->schema) {
+    g_object_unref(priv->schema);
+    priv->schema = nullptr;
+  }
+
+  G_OBJECT_CLASS(garrow_json_read_options_parent_class)->dispose(object);
+}
+
+static void
+garrow_json_read_options_set_property(GObject *object,
+                                      guint prop_id,
+                                      const GValue *value,
+                                      GParamSpec *pspec)
+{
+  auto priv = GARROW_JSON_READ_OPTIONS_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_JSON_READER_USE_THREADS:
+    priv->read_options.use_threads = g_value_get_boolean(value);
+    break;
+  case PROP_JSON_READER_BLOCK_SIZE:
+    priv->read_options.block_size = g_value_get_int(value);
+    break;
+  case PROP_JSON_READER_ALLOW_NEWLINES_IN_VALUES:
+    priv->parse_options.newlines_in_values = g_value_get_boolean(value);
+    break;
+  case PROP_JSON_READER_UNEXPECTED_FIELD_BEHAVIOR:
+    priv->parse_options.unexpected_field_behavior =
+      static_cast<arrow::json::UnexpectedFieldBehavior>(g_value_get_enum(value));
+    break;
+  case PROP_JSON_READER_SCHEMA:
+    priv->schema = GARROW_SCHEMA(g_value_dup_object(value));
+    priv->parse_options.explicit_schema = garrow_schema_get_raw(priv->schema);
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_json_read_options_get_property(GObject *object,
+                                      guint prop_id,
+                                      GValue *value,
+                                      GParamSpec *pspec)
+{
+  auto priv = GARROW_JSON_READ_OPTIONS_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_JSON_READER_USE_THREADS:
+    g_value_set_boolean(value, priv->read_options.use_threads);
+    break;
+  case PROP_JSON_READER_BLOCK_SIZE:
+    g_value_set_int(value, priv->read_options.block_size);
+    break;
+  case PROP_JSON_READER_ALLOW_NEWLINES_IN_VALUES:
+    g_value_set_boolean(value, priv->parse_options.newlines_in_values);
+    break;
+  case PROP_JSON_READER_UNEXPECTED_FIELD_BEHAVIOR:
+    g_value_set_enum(value, static_cast<int>(priv->parse_options.unexpected_field_behavior));
+    break;
+  case PROP_JSON_READER_SCHEMA:
+    if (priv->schema) {
+      g_value_set_object(value, priv->schema);
+    } else {
+      g_value_set_object(value, nullptr);
+    }
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_json_read_options_init(GArrowJSONReadOptions *object)
+{
+  auto priv = GARROW_JSON_READ_OPTIONS_GET_PRIVATE(object);
+  priv->pool = arrow::default_memory_pool();
```

Comment:
Ah, `priv->pool` is leaked.
We need to execute `priv->poll = nullptr;` on `_dispose()` for this.
Or we can remove `priv->pool` at all.
I think that the latter is better. What do you think about this?

Reply:
I think so too. I've removed `priv-pool`.